src/middlewared/middlewared/alert/source/certificates.py

```diff
@@ -64,7 +64,10 @@ async def check(self):
                 # check the parsed certificate(s) for expiration
                 if cert['cert_type'] == 'CERTIFICATE':
                     diff = (datetime.strptime(cert['until'], '%a %b %d %H:%M:%S %Y') - utc_now()).days
-                    if diff < 10:
+                    # For renewable certificates (those with renew_days), only alert if within renewal window
+                    # For non-renewable certificates, use the default 10-day threshold
+                    alert_threshold = cert.get('renew_days', 10) - 1 if cert.get('renew_days') is not None else 10
```

Comment:
The logic is insanely hard to comprehend. If the `renew_days` key isn't present, default to 10. Else it's `renew_days` - 1 OR it's `10` by default if `renew_days` is in the dictionary but is a falsey object??